Lesson18/main.js

```diff
@@ -64,11 +64,33 @@ const init = async () => {
     initArrowButtons();
     initIndicator(imgData.length);
     slideWrapper.appendChild(createCounterOfSlide(imgData));
+    autoPlay.start();
   }
 }
 
 init();
 
+/**
+ * index of the element being displayed.
+ * @type {Number}
+ */
+let currentIndex = 0;
+
+let autoPlayerId;
+const autoPlay = {
+  start: function () {
+    autoPlayerId = setInterval(() => {
+      const slideItems = [...document.querySelectorAll(".slide-item")];
+      currentIndex < slideItems.length - 1 ? ++currentIndex : currentIndex = 0;
+      playOfSlideshow();
+    }, 3000);
+  },
+  reset: function () {
+    clearInterval(autoPlayerId);
+    this.start();
+  }
+}
```

Comment:
I thought it was brilliant that you put the function in the value of the object.
So cool!!!

Reply:
Thank you for everything.✨

Reply:
I think it's cool too!